File: projects/code_combat/7_Sarven_Desert/339-Pet_Engineer/pet_engineer.py
def onHear(event):
    archer = pet.findNearestByType("archer")
    soldier = pet.findNearestByType("soldier")
    if event.speaker == archer:
        pet.moveXY(32, 30)
    elif event.speaker == "soldier":
        pet.moveXY(48, 30)

pet.on("hear", onHear)

while True:
    enemy = hero.findNearestEnemy()

    # fix for running away hero
    if enemy && hero.distanceTo(enemy) < 15:
        hero.attack(enemy)
    else:
        hero.moveXY(60, 15)

File: projects/code_combat/7_Sarven_Desert/339-Pet_Engineer/petEngineer.js
// Move your pet to the left or right button as needed.

function onHear(event) {
    // Find the guards to listen to.
    var archer = pet.findNearestByType("archer");
    var soldier = pet.findNearestByType("soldier");
    // If event.speaker is the `archer`:
    if (event.speaker == archer) {
        // Move to the left button.
        pet.moveXY(32, 30);
    }
    // If event.speaker is the `soldier`:
    else if (event.speaker == soldier) {
        // Move to the right button.
        pet.moveXY(48, 30);
    }
}

pet.on("hear", onHear);

// You don't have to change the code below.
// Your hero should protect the bottom right passage.
while(true) {
    var enemy = hero.findNearestEnemy();

    // fix for running away hero
    if (enemy && hero.distanceTo(enemy) < 10)
        hero.attack(enemy);
    else
        hero.moveXY(60, 15);
}
